src/analyze/analyze-security.c

```diff
@@ -513,10 +524,10 @@ static int assess_system_call_architectures(
         assert(ret_badness);
         assert(ret_description);
 
-        if (strv_isempty(info->system_call_architectures)) {
+        if (set_isempty(info->system_call_architectures)) {
                 b = 10;
                 d = strdup("Service may execute system calls with all ABIs");
-        } else if (strv_equal(info->system_call_architectures, STRV_MAKE("native"))) {
+        } else if (set_contains(info->system_call_architectures, "native")) {
```

Comment:
this isn't right. you are changing the definition of this test here completely. previously if you have "native" and "i386" in the list this would be detected as problematic, since a non-native arch is allowed. after your change that's not the case anymore, suddenly that would be ok